### ISSUE TITLE
issue decrypting for non-first key if all use the same pass phrase

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#999]
> You may be aware of this and consider it a feature, but I find it a bit strange.  If one has three private keys associated with an email address in FlowCrypt, clicking the "enter passphrase" link opens a passphrase-entry window listing all three keys.  So far, so good.  If all three keys have the same passphrase, entering it immediately decrypts the ciphertext _if_ the message is encrypted to the first (primary) key.  If, however, the message is encrypted to the second key, entering the passphrase seemingly does nothing.  Clicking the entry link again opens the window with the second and third keys listed.  Re-entering the passphrase decrypts the email, but if the email is encrypted to the third key, three tries is required.
> 
> It should be possible for FlowCrypt to ascertain which key a message is encrypted to.  GnuPG does.  Failing that, why not cache the passphrase briefly and retry until it hits the correct key.  Either of these solutions should also work if the keys have different passphrases.

**Comment by tomholub** [comment]
ref https://mail.google.com/mail/u/0/#inbox/1618b4487c61b503

**Comment by tomholub** [comment]
This looks interesting, please also try to come up with a test case for it.

**Comment by michael-volynets** [comment]
I couldn't reproduce this issue,
The way I tried to reproduce this:

1. I have an account on FlowCrypt and I added 3 private keys with **the same** passphrase generated in GnuPG software.
2. I checked `Require passphrase to open encrypted email` checkbox in FlowCrypt Settings tab to see that `enter passphrase` button described in the description.
3. I sent 3 encrypted by each key emails to this account.

Then when I try to open the first email it shows "enter passphrase" button, I enter the PP (main private key) and the message is successfully decrypted but when I open the 2-nd and 3-rd message - FlowCrypt decrypts them without asking PP.

@tomholub Is this correct behavior? I mean when `Require passphrase to open encrypted email` checkbox is checked, it only requires the PP for messages that can be decrypted using main key. And the messages that can be decrypted using a 2-nd or 3-rd key don't require to enter PP.
